dts/bindings/sensor/st,stm32-dts.yaml

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2024, Aurelien Jarno
+# SPDX-License-Identifier: Apache-2.0
+
+description: STM32 family Digital Temperature Sensor node
```

Comment:
Right now there's stm32_temp driver in-tree (based on ADC readings).

Is there a way we could abstract to the user which implementation is selected? I'm thinking of Kconfig selection based on the presence of this DTS node. That way, the user doesn't really has to _choose_ between the two: if it's digital use that one.

Reply:
I have mixed feeling about that. Users might have reasons to choose about that. While it is a simpler driver than the analog one with the ADC, it is less accurate and use ~10 times more power (excluding the ADC consumption, but users might have to use the ADC anyway).

Reply:
> Right now there's stm32_temp driver in-tree (based on ADC readings).
> 
> Is there a way we could abstract to the user which implementation is selected? I'm thinking of Kconfig selection based on the presence of this DTS node. That way, the user doesn't really has to _choose_ between the two: if it's digital use that one.

The only way I can see to abstract that given they are two different peripherals with different device tree properties is to remove the analog sensor from the device tree for the STM32 devices having a digital sensor, and point the `die_temp` label to it. This assumes users are referring to it via label and not via name or compat.

Reply:
> Users might have reasons to choose about that. While it is a simpler driver than the analog one with the ADC, it is less accurate and use ~10 times more power (excluding the ADC consumption, but users might have to use the ADC anyway).

That would be worth documenting in this description section. Once the 2 bindings will be available, they'll need element to help in deciding which one to choose.